cli/createTemplate.sh

```diff
@@ -876,23 +876,25 @@ function process_template() {
 
     # Cleanup output directory
     if [[ "${deployment_unit_state_subdirectories}" == "true" ]]; then
-
-      # Remove existing files for the current level being careful to preserve stacks
-      readarray -t existing_files < <(find "${cf_dir}" -mindepth 1 -maxdepth 1 -type f \
-      \(  -name "${cleanup_level}-*" \
-          -and -not -name "${cleanup_level}-*-stack.json" \
-          -and -not -name "${cleanup_level}-*-lastchange.json" \) )
-
-      for e in "${existing_files[@]}"; do
-        local existing_filename="$(fileName "${e}")"
-        debug "... checking file ${existing_filename} ..."
-        # If generated, then ignore
-        $(inArray "results_list" "${existing_filename}") && continue
-
-        # Wasn't generated so remove
-        info "... removing ${existing_filename} ..."
-        rm  -f "${cf_dir}/${existing_filename}"
-      done
+      # Don't cleanup output directory when compiling test cases.
+      if [[ ! "${GENERATION_INPUT_SOURCE}" == "mock" ]]; then 
```

Comment:
I think this check would be on `OUTPUT_DIR` rather than the input source being mock. The outdir parameter is what is dumping all the data in one spot, not necessarily the mock input source

Reply:
Or maybe this should be an extra param ( CLEANUP ) which is enabled by default

Reply:
I've just gone with the OUTPUT_DIR for now. We can add cleanup later if we find it's needed elsewhere.